### PR TITLE
async/await check_password

### PR DESCRIPTION
**Pull request by jmastr**
Was introduced in Synapse v1.15.0.

This may be related to:
* https://github.com/ma1uta/matrix-synapse-rest-password-provider/issues/1

### CODE REVIEW

File: rest_auth_provider.py [hunk 1]
```diff
@@ -83,14 +82,14 @@ def check_password(self, user_id, password):
 
             # fixme: temporary fix
             try:
-                store = yield self.account_handler._hs.get_profile_handler().store  # for synapse >= 1.9.0
+                store = await self.account_handler._hs.get_profile_handler().store  # for synapse >= 1.9.0
```
Comment (jmastr) [inline rest_auth_provider.py:85]: @ma1uta I am not sure about this whole `try... except` part. The async/await was introduced in Synapse v1.15.0.

**Comment by ma1uta** [comment]
Thanks for PR!

**Comment by davidmehren** [comment]
As discovered in #4, this PR seems to break authentication :cry:

**Comment by ma1uta** [comment]
Reverted in the https://github.com/ma1uta/matrix-synapse-rest-password-provider/commit/c782c84aeab1872e73b6c29aadb99d3852e26bbd

**Pull request merged**